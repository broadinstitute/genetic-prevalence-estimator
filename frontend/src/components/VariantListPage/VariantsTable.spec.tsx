import { render } from "@testing-library/react";

import { variantListFactory } from "../../../tests/__factories__/VariantList";

import VariantsTable from "./VariantsTable";

describe("Variant Table", () => {
  it("has no unexpected changes", () => {
    const testVariantList = variantListFactory.build();
    const result = render(
      <VariantsTable
        includePopulationFrequencies={[]}
        variantList={testVariantList}
        selectedVariants={new Set(["12-123-A-C", "12-234-A-C"])}
        shouldShowVariant={() => true}
        variantNotes={{ "12-123-A-C": "note for test variant 12-123-A-C" }}
        onChangeSelectedVariants={() => {}}
        onEditVariantNote={() => {}}
      />
    );
    expect(result.asFragment()).toMatchSnapshot();
  });
});
