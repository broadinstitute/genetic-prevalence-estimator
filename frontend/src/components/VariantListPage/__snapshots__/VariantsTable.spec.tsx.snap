// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variant Table has no unexpected changes 1`] = `
<DocumentFragment>
  <table
    class="chakra-table css-tl2dr7"
    role="table"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <th
          class="css-0"
          scope="col"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
              data-indeterminate=""
            >
              <div
                style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
              >
                <svg
                  class="css-f5ntj6"
                  style="stroke: currentColor; stroke-width: 4; opacity: 1; transform: none;"
                  viewBox="0 0 24 24"
                >
                  <line
                    x1="21"
                    x2="3"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </div>
            </span>
            <span
              class="chakra-checkbox__label css-1oeb2oe"
            >
              <span
                class="css-f8n5zr"
              >
                Include variants in calculations
              </span>
            </span>
          </label>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          Note
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Variant ID
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Gene
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Transcript
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            VEP consequence
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            LOFTEE
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            HGVSc
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            HGVSp
          </button>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Clinical significance
          </button>
        </th>
        <th
          class="css-0"
          data-is-numeric="true"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Allele count
          </button>
        </th>
        <th
          class="css-0"
          data-is-numeric="true"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Allele number
          </button>
        </th>
        <th
          aria-sort="descending"
          class="css-0"
          data-is-numeric="true"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Allele frequency
          </button>
          <span
            style="position: absolute; right: 3px; top: calc(50% - 10px);"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                fill="currentColor"
              />
            </svg>
          </span>
        </th>
        <th
          class="css-0"
          scope="col"
          style="position: relative;"
        >
          <button
            style="display: block; width: 100%; height: 100%; appearance: none; font-weight: inherit; text-align: inherit;"
          >
            Source
          </button>
        </th>
      </tr>
    </thead>
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
            />
            <span
              class="chakra-checkbox__label css-1oeb2oe"
            >
              <span
                class="css-f8n5zr"
              >
                Include this variant in calculations
              </span>
            </span>
          </label>
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <th
          class="css-1l7mgse"
          role="gridcell"
          scope="row"
        >
          <span
            style="display: inline-block; max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          >
            <a
              class="chakra-link css-0"
              href="https://gnomad.broadinstitute.org/variant/12-567-A-C?dataset=gnomad_r2_1"
              rel="noopener noreferrer"
              target="_blank"
            >
              12-567-A-C
            </a>
          </span>
        </th>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          <span
            class="css-1bvc4cc"
          >
            <span>
              0
            </span>
          </span>
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          Custom
        </td>
      </tr>
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
            />
            <span
              class="chakra-checkbox__label css-1oeb2oe"
            >
              <span
                class="css-f8n5zr"
              >
                Include this variant in calculations
              </span>
            </span>
          </label>
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <th
          class="css-1l7mgse"
          role="gridcell"
          scope="row"
        >
          <span
            style="display: inline-block; max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          >
            <a
              class="chakra-link css-0"
              href="https://gnomad.broadinstitute.org/variant/12-456-A-C?dataset=gnomad_r2_1"
              rel="noopener noreferrer"
              target="_blank"
            >
              12-456-A-C
            </a>
          </span>
        </th>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          <span
            class="css-1bvc4cc"
          >
            <span>
              0
            </span>
          </span>
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          Custom
        </td>
      </tr>
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
            />
            <span
              class="chakra-checkbox__label css-1oeb2oe"
            >
              <span
                class="css-f8n5zr"
              >
                Include this variant in calculations
              </span>
            </span>
          </label>
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <th
          class="css-1l7mgse"
          role="gridcell"
          scope="row"
        >
          <span
            style="display: inline-block; max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          >
            <a
              class="chakra-link css-0"
              href="https://gnomad.broadinstitute.org/variant/12-345-A-C?dataset=gnomad_r2_1"
              rel="noopener noreferrer"
              target="_blank"
            >
              12-345-A-C
            </a>
          </span>
        </th>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          <span
            class="css-1bvc4cc"
          >
            <span>
              0
            </span>
          </span>
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          Custom
        </td>
      </tr>
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              checked=""
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
              data-checked=""
            >
              <div
                style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
              >
                <svg
                  class="css-f5ntj6"
                  style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                  viewBox="0 0 12 10"
                >
                  <polyline
                    points="1.5 6 4.5 9 10.5 1"
                  />
                </svg>
              </div>
            </span>
            <span
              class="chakra-checkbox__label css-1oeb2oe"
              data-checked=""
            >
              <span
                class="css-f8n5zr"
              >
                Include this variant in calculations
              </span>
            </span>
          </label>
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <th
          class="css-1l7mgse"
          role="gridcell"
          scope="row"
        >
          <span
            style="display: inline-block; max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          >
            <a
              class="chakra-link css-0"
              href="https://gnomad.broadinstitute.org/variant/12-234-A-C?dataset=gnomad_r2_1"
              rel="noopener noreferrer"
              target="_blank"
            >
              12-234-A-C
            </a>
          </span>
        </th>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          <span
            class="css-1bvc4cc"
          >
            <span>
              0
            </span>
          </span>
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          Custom
        </td>
      </tr>
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          <label
            class="chakra-checkbox css-1uiwwan"
          >
            <input
              checked=""
              class="chakra-checkbox__input"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value=""
            />
            <span
              aria-hidden="true"
              class="chakra-checkbox__control css-dnty2r"
              data-checked=""
            >
              <div
                style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
              >
                <svg
                  class="css-f5ntj6"
                  style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                  viewBox="0 0 12 10"
                >
                  <polyline
                    points="1.5 6 4.5 9 10.5 1"
                  />
                </svg>
              </div>
            </span>
            <span
              class="chakra-checkbox__label css-1oeb2oe"
              data-checked=""
            >
              <span
                class="css-f8n5zr"
              >
                Include this variant in calculations
              </span>
            </span>
          </label>
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <button
            aria-label="View note"
            class="chakra-button css-1jyk1h1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="2"
                />
              </g>
            </svg>
          </button>
        </td>
        <th
          class="css-1l7mgse"
          role="gridcell"
          scope="row"
        >
          <span
            style="display: inline-block; max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          >
            <a
              class="chakra-link css-0"
              href="https://gnomad.broadinstitute.org/variant/12-123-A-C?dataset=gnomad_r2_1"
              rel="noopener noreferrer"
              target="_blank"
            >
              12-123-A-C
            </a>
          </span>
        </th>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          <span
            style="display: inline-block; max-width: 150px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
          />
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        />
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          <span
            class="css-1bvc4cc"
          >
            <span>
              0
            </span>
          </span>
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          data-is-numeric="true"
          role="gridcell"
        >
          0
        </td>
        <td
          class="css-1l7mgse"
          role="gridcell"
        >
          Custom
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
